src/main/java/org/jenkinsci/plugins/mesos/MesosSlaveInfo.java

```diff
@@ -0,0 +1,186 @@
+package org.jenkinsci.plugins.mesos;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.model.Node;
+import java.util.Collections;
+import java.util.List;
+import org.apache.mesos.Protos.ContainerInfo.DockerInfo.Network;
+
+/**
+ * This POJO describes a Jenkins agent for Mesos on 0.x and 1.x of the plugin. It is used to migrate
+ * older configurations to {@link MesosAgentSpecTemplate} during deserialization. See {@link
+ * MesosCloud#readResolve()} for the full migration.
+ */
+public class MesosSlaveInfo {
+
+  //  ???       <nodeProperties/>
+  private transient Node.Mode mode;
+  private transient String labelString;
+  private transient Double slaveCpus;
+  private transient Double diskNeeded;
+  private transient int slaveMem;
+
+  @SuppressFBWarnings("UUF_UNUSED_FIELD")
+  private transient Double executorCpus;
+
+  private transient int minExecutors;
+  private transient int maxExecutors;
+  private transient int executorMem;
+
+  @SuppressFBWarnings("UUF_UNUSED_FIELD")
+  private transient String remoteFSRoot;
+
+  private transient int idleTerminationMinutes;
+  private transient String jvmArgs;
+  private transient String jnlpArgs;
+  private transient boolean defaultSlave;
+
+  @SuppressFBWarnings("UUF_UNUSED_FIELD")
+  private transient List<URI> additionalURIs;
+
+  private transient ContainerInfo containerInfo;
+
+  /**
+   * Resolves the old agent configuration after deserialization.
+   *
+   * @return the agent configt as a {@link MesosAgentSpecTemplate}.
+   */
+  private Object readResolve() {
+
+    // Migrate to 2.x spec template
+    return new MesosAgentSpecTemplate(
+        this.labelString,
+        this.mode,
+        this.slaveCpus.toString(),
+        Integer.toString(this.slaveMem),
+        this.idleTerminationMinutes,
+        this.minExecutors,
+        this.maxExecutors,
+        this.diskNeeded.toString(),
+        this.executorMem,
+        "",
+        this.jvmArgs,
+        this.jnlpArgs,
+        this.defaultSlave,
+        "", // TODO: support additional URIs in MesosAgentSpecTemplate
+        this.containerInfo.dockerImage);
+  }
+
+  public static class URI {
+
+    private final String value;
+    private final boolean executable;
+    private final boolean extract;
+
+    public URI(String value, boolean executable, boolean extract) {
+      this.value = value;
+      this.executable = executable;
+      this.extract = extract;
+    }
+
+    public String getValue() {
+      return value;
+    }
+  }
+
+  public static class ContainerInfo {
+
+    private final String type;
+    private final String dockerImage;
+    private final List<Volume> volumes;
+    private final List<Parameter> parameters;
+    private final String networking;
```

Comment:
Could we use the Network-Enum directly here instead of a string? Or does that clash somewhere else?

Reply:
Oh, they're mapped from the XML, right? Still, maybe we can keep it a string in the constructor and convert it directly into the enum?

Reply:
We can and should. This is a verbatim copy from the original 0.x and 1.x config. I'd convert them to an enum once we use them.